src/runtime/format.js

```diff
@@ -10,9 +10,41 @@
  * governing permissions and limitations under the License.
  */
 
+const moment = require('moment');
+const numeral = require('numeral');
+require('numeral/locales');
+
+function parseDate(format, locale, timezone) {
+  const date = moment(parseInt(format, 10));
+  if (date.isValid()) {
+    return date.utc(timezone).locale(locale);
+  }
+
+  return null;
+}
+
 module.exports = function format(fmt, args) {
-  const argArray = Array.isArray(args) ? args : [args];
-  return fmt.replace(/{(\d+)}/g, (match, number) => (typeof argArray[number] !== 'undefined'
-    ? argArray[number]
-    : match));
+  const PLACEHOLDER_REGEX = /{(\d+)}/g;
+  const NUMBER_FORMAT_REGEX = /^[0-9.,#;$E%()-]*$/gm;
+  const PLACEHOLDER_NUMBER = /#/g;
+  const argArray = Array.isArray(args.format) ? args.format : [args.format];
+  const locale = args.locale ? args.locale : '';
+  const timezone = args.timezone ? args.timezone.replace('GMT', '').replace('UTC', '') : '';
+
+  if (NUMBER_FORMAT_REGEX.test(fmt)) {
+    // apply locale
+    numeral.locale(locale.toLowerCase());
+    const number = numeral(args.format).format(fmt.replace(PLACEHOLDER_NUMBER, '0'));
+    // reset locale
+    numeral.reset();
```

Comment:
I don't like this too much, as it sets the `locale` globally. is there a way to create a local instance or use a different library?

It might not be a problem, since the function is synchronous. but if there isn't a better way, at least to a 

```
try {
...
} finally {
  numeral.reset();
}
```

Reply:
I did have a look and apparently this is the only way to set the locale for numeral but I agree it's not great. It seems like this is the most widely used numeral formatting library for js/node so I went with that one.

I'll wrap it up in a try/finally block - that will at least guarantee the reset happens.